lib/internal/Magento/Framework/View/Template/Html/Minifier.php

```diff
@@ -116,49 +117,94 @@ public function minify($file)
         $dir = dirname($file);
         $fileName = basename($file);
         $content = $this->readFactory->create($dir)->readFile($fileName);
-        //Storing Heredocs
-        $heredocs = [];
-        $content = preg_replace_callback(
-            '/<<<([A-z]+).*?\1;/ims',
-            function ($match) use (&$heredocs) {
-                $heredocs[] = $match[0];
+        $heredocs = null;
 
-                return '__MINIFIED_HEREDOC__' .(count($heredocs) - 1);
-            },
+        // Safely minify PHP code and remove single-line PHP comments by using a parser.
+        if (null !== $content) {
+            $parser = (new \PhpParser\ParserFactory())->create(\PhpParser\ParserFactory::PREFER_PHP7);
+
+            /**
+             * Prevent problems with deeply nested ASTs if Xdebug is enabled.
+             * @see https://github.com/nikic/PHP-Parser/blob/v4.4.0/doc/2_Usage_of_basic_components.markdown#bootstrapping
+             */
+            $nestingLevelConfigValue = ini_get('xdebug.max_nesting_level');
+
+            if (false !== $nestingLevelConfigValue) {
+                ini_set('xdebug.max_nesting_level', '3000');
+            }
+
+            try {
+                $ast = $parser->parse($content);
+
+                $traverser = new \PhpParser\NodeTraverser();
+                $traverser->addVisitor(new Php\NodeVisitor());
+                $ast = $traverser->traverse($ast);
+
+                $prettyPrinter = new Php\PrettyPrinter();
+                $content = $prettyPrinter->prettyPrintFile($ast);
+                $heredocs = $prettyPrinter->getDelayedHeredocs();
+            } catch (\Error $error) {
+                // Some PHP code is seemingly invalid, or too complex.
```

Comment:
In this case doesn't it make more sense to just log a warning and return the unminified code instead of guessing that you could do a better job than the PHP parser at minifying some mangled script?